obspy/io/gcf/core.py

```diff
@@ -180,49 +181,138 @@ def merge_gcf_stream(st):
     return Stream(traces=traces)
 
 
-def _is_gcf(filename):
+def _is_gcf(file):
     """
     Checks whether a file is GCF or not.
 
-    :type filename: str
-    :param filename: path to GCF file to be checked.
+    :type file: str or file-like object
+    :param file: path to GCF file to be checked
     :rtype: bool
     :return: True if a object pointed to by path is a GCF file.
     """
-    if not os.path.isfile(filename) or os.path.getsize(filename) % 1024:
-        # File either does not point at a file object or file is not of
-        # proper size
-        return False
-
     # Load shared library
     gcf_io = _load_cdll("gcf")
 
     # declare function argument and return types
-    gcf_io.read_gcf.argtypes = [ctypes.c_char_p,
-                                ctypes.POINTER(_GcfFile),
-                                ctypes.c_int]
-    gcf_io.read_gcf.restype = ctypes.c_int
     gcf_io.free_GcfFile.argtypes = [ctypes.POINTER(_GcfFile)]
     gcf_io.free_GcfFile.restype = None
 
+    obj = None
     try:
         # Decode first block
         obj = _GcfFile()
-        b_filename = filename.encode('utf-8')
-        ret = gcf_io.read_gcf(b_filename, obj, 3)
+        ret = _fill_gcf_from_file(file, obj, 3)
```

Comment:
Considering that I am a little rusty on C and have no expertise on Gcf formats, as you said the priority was to change as little as possible rather than being Pythonic (for instance, that function is intended to mimic the C counterpart) and also make life easier for reviewers when comparing with old implementation (see discussion with @paitor above, but in general if you guys know other potential experts on the subject please invite them to get their feedback)